transit/script/bart.py

```diff
@@ -0,0 +1,396 @@
+            "RICH",
+            "ORIN",
+            "LAFY",
+            "WCRK",
+            "CONC",
+            "NCON",
+            "PITT",
+            "ANTC",
+            "DELN",
+            "PHIL",
+            "PCTR",
+            "PLZA",
+        ],
+        "Alameda": [
+            "WOAK",
+            "12TH",
+            "19TH",
+            "MCAR",
+            "ASHB",
+            "DUBL",
+            "WDUB",
+            "CAST",
+            "WARM",
+            "UCTY",
+            "SHAY",
+            "HAYW",
+            "BAYF",
+            "SANL",
+            "OAKL",
+            "COLS",
+            "FTVL",
+            "LAKE",
+            "ROCK",
+            "DBRK",
+            "NBRK",
+            "FRMT",
+        ],
+        "Santa Clara": [],  # Add stations for Santa Clara if available
+    }
+    for county, stations in counties.items():
+        if station in stations:
+            return county
+    return None
+
+
+def process_BART_county(
+    file_name, model_run_dir, output_dir, model_BART_county, model_BART
+):
+    BART_county = process_BART_data(file_name, model_run_dir, output_dir, model_BART)
+
+    # Add the 'County' column to the DataFrame
+    BART_county["County"] = BART_county["Station"].apply(
+        lambda x: map_station_to_county(x)
+    )
+    BART_county = (
+        BART_county.groupby(["County", "TOD"])[["Boardings", "Alightings"]]
+        .sum()
+        .reset_index()
+    )
+    BART_county.to_csv(output_dir / model_BART_county, index=False)
+
+
+# BART Screenline
+def process_BART_SL_data(file_name, A, B):
+    # Read the DBF file and group by 'A' and 'B' while summing 'AB_VOL'
+    BART_SL = read_dbf_and_groupby_sum(file_name, "BART", ["A", "B", "TOD"], "AB_VOL")
+
+    # Filter rows for IB (16510 to 16511)
+    IB = BART_SL[(BART_SL["A"] == A) & (BART_SL["B"] == B)].copy()
+    IB["Direction"] = "IB"
+
+    # Filter rows for OB (16511 to 16510)
+    OB = BART_SL[(BART_SL["A"] == B) & (BART_SL["B"] == A)].copy()
+    OB["Direction"] = "OB"
+
+    # Concatenate IB and OB DataFrames
+    result = pd.concat([IB, OB])
+
+    return result
+
+
+def BART_SL_Concat(file_name, A, B, Screenline):
+    # Concatenate the DataFrames
+    BART_SL = process_BART_SL_data(file_name, A, B)
+
+    # Add the 'Screenline' column with 'Countyline'
+    BART_SL["Screenline"] = Screenline
+
+    # Create the 'Key' column by combining 'Screenline', 'Direction', and 'TOD'
+    BART_SL["Key"] = BART_SL["Screenline"] + BART_SL["Direction"] + BART_SL["TOD"]
+
+    # Rename the 'AB_VOL' column to 'Ridership'
+    BART_SL = BART_SL.rename(columns={"AB_VOL": "Ridership"})
+
+    # Select and reorder columns
+    BART_SL = BART_SL[["Screenline", "Direction", "TOD", "Key", "Ridership"]]
+
+    # Sort by 'Key' and reset the index
+    BART_SL = BART_SL.sort_values(by="Key").reset_index(drop=True)
+
+    return BART_SL
+
+
+station_locations = {
+    "downtown": ["CIVC", "POWL", "MONT", "EMBR"],
```

Comment:
@sdrewc @dantischler do we have better names than "downtown" vs "not_downtown"?
"market_st" vs "ex_market_st"?